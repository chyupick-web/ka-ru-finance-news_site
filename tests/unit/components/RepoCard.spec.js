import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/RepoCard.vue'
import BootstrapVue from 'bootstrap-vue'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

describe('RepoCard.vue', () => {
  it('matches snapshot with default props', () => {
    const wrapper = shallowMount(Component, { localVue })
    expect(wrapper.html()).toMatchSnapshot('defaultProps')
  })

  it('matches snapshot with set props', () => {
    const propsData = {
      item: {
        owner: '<OWNER>',
        ownerUrl: 'https://github.com/<OWNER>',
        name: '<NAME>',
        url: 'https://github.com/<OWNER>/<NAME>',
        description: 'Nulla quis lorem ut libero malesuada feugiat.',
        stars: 150,
        forks: 25,
        language: {
          title: 'Python',
          color: '#3572A5'
        }
      }
    }
    const wrapper = shallowMount(Component, { localVue, propsData })
    expect(wrapper.html()).toMatchSnapshot('withProps')
  })
})
