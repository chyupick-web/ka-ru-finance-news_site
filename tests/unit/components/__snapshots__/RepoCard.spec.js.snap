// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard.vue matches snapshot with default props: defaultProps 1`] = `
<div class="border border-sei-gray-lightest bg-white p-3 h-100">
  <h4><a href="#" target="_self" class="text-cmu-red font-weight-bold"></a> <span class="d-inline-block ml-1 mr-1">/</span> <a href="#" target="_self" class="text-cmu-red"></a></h4>
  <p class="small"></p>
  <div class="info-section"><span class="d-inline-block mr-3 font-weight-bold"><svg><use xlink:href="#star"></use></svg>
      0
    </span> <span class="d-inline-block mr-3 font-weight-bold"><svg><use xlink:href="#fork"></use></svg>
      0
    </span> <span class="d-inline-block font-weight-bold"><i aria-hidden="true" class="d-inline-block rounded-circle" style="background-color: rgb(255, 255, 255);"></i>
      
    </span></div>
</div>
`;

exports[`RepoCard.vue matches snapshot with set props: withProps 1`] = `
<div class="border border-sei-gray-lightest bg-white p-3 h-100">
  <h4><a href="https://github.com/<OWNER>" target="_self" class="text-cmu-red font-weight-bold">&lt;OWNER&gt;</a> <span class="d-inline-block ml-1 mr-1">/</span> <a href="https://github.com/<OWNER>/<NAME>" target="_self" class="text-cmu-red">&lt;NAME&gt;</a></h4>
  <p class="small">Nulla quis lorem ut libero malesuada feugiat.</p>
  <div class="info-section"><span class="d-inline-block mr-3 font-weight-bold"><svg><use xlink:href="#star"></use></svg>
      150
    </span> <span class="d-inline-block mr-3 font-weight-bold"><svg><use xlink:href="#fork"></use></svg>
      25
    </span> <span class="d-inline-block font-weight-bold"><i aria-hidden="true" class="d-inline-block rounded-circle" style="background-color: rgb(53, 114, 165);"></i>
      Python
    </span></div>
</div>
`;
